### PR TITLE
DB-9867: make sure to raise an exception if unique constraint is violated for system tables

### PR DESCRIPTION
**Pull request by jyuanca**

### CODE REVIEW

**Comment by jyuanca** [comment]
Please also review ee branch

**Comment by jyuanca** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1086/

**Comment by jyuanca** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1090/

**Comment by jyuanca** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java [hunk 1]
```diff
@@ -1183,14 +1183,20 @@ public boolean existsSchemaOwnedBy(String authid,TransactionController tc) throw
     public void addColumnStatistics(ExecRow row,
                               TransactionController tc) throws StandardException{
         TabInfoImpl ti=getNonCoreTI(SYSCOLUMNSTATS_CATALOG_NUM);
-        ti.insertRow(row,tc);
+        int insertRetCode = ti.insertRow(row,tc);
+        if (insertRetCode != TabInfoImpl.ROWNOTDUPLICATE)
+            throw StandardException.newException(SQLState.LANG_DUPLICATE_KEY_CONSTRAINT,
+                    "SYSCOLUMNSTATISTICS_INDEX1", SYSCOLUMNSTATISTICSRowFactory.TABLENAME_STRING);
```
Comment (hatyo) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java:1189]: you could refactor this in a method called `insertOrThrow` and reuse it everywhere :)

Comment (jyuanca) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java:1189]: The thrown exception, number of parameter, and error code are not the same. Need to write at least three versions of insertOrThrow. I may do this later if that can make code much cleaner.

**Pull request merged**